platform/src/main/java/org/hillview/storage/CassandraDatabase.java

```diff
@@ -0,0 +1,264 @@
+
+    public void closeClusterConnection() {
+        if (this.session != null)
+            this.session.close();
+        if (this.cluster != null)
+            this.cluster.close();
+    }
+
+    /** Connection for running server-side query. */
+    private void connectLocalProbe() {
+        try {
+            if (Utilities.isNullOrEmpty(info.user)) {
+                this.probe = nodeProbeFactory.create(info.host, info.jmxPort);
+            } else
+                this.probe = nodeProbeFactory.create(info.host, info.jmxPort,
+                    info.user, info.password);
+        } catch (IOException | SecurityException e) {
+            Throwable rootCause = Throwables.getRootCause(e);
+            HillviewLogger.instance.error("Nodetool failed to connect", info.host + ":" + info.jmxPort +
+                " - " + rootCause.getClass().getSimpleName() + " : " + rootCause.getMessage());
+            throw new RuntimeException(e);
+        }
+    }
+
+    public static class TokenRange {
+        public String start_token;
+        public String end_token;
+        public List<String> endpoints = new ArrayList<>();
+
+        public TokenRange(String rawEntry) {
+            try{
+                String[] arr = rawEntry.split(":");
+                this.start_token = arr[1].split(",")[0];
+                this.end_token = arr[2].split(",")[0];
+
+                String rawEndpoints = arr[3].replace(", rpc_endpoints", "");
+                rawEndpoints = rawEndpoints.substring(1,rawEndpoints.length()-1);
+                String[] ips = rawEndpoints.split(",");
+                for (String ip : ips) {
+                    endpoints.add(new String(ip));
+                }
+            } catch (Exception e) {
+                throw new RuntimeException("Failed to parse token range", e);
+            }
+        }
+
+        public String toString() {
+            return "start: " + this.start_token + ", end: " + this.end_token +
+                ", endpoints: " + this.endpoints.toString();
+        }
+    }
+
+    /**
+     * Shows key-range (table partition) and endpoints (replica) which stores each key-range.
+     * This key-range distribution is the same on every node in the cluster.
+     */
+    private void setTokenRanges() {
+        String keyspace = this.info.database;
+        try {
+            TokenRange tr;
+            for (String tokenRangeString : this.probe.describeRing(keyspace)) {
+                tr = new TokenRange(tokenRangeString);
+                this.tokenRanges.add(tr);
+            }
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    public static class CassTable {
+        public String keyspace;
+        public List<String> tables;
+
+        public CassTable(String keyspace, List<String> tables) {
+            this.keyspace = keyspace;
+            this.tables = tables;
+        }
+
+        public String toString(){
+            String result = this.keyspace;
```

Comment:
space. Also, don't use String concatenation in a loop, use a StringBuilder instead.